addons/uom/static/src/components/many2x_uom_tags/many2x_uom_tags.js

```diff
@@ -10,25 +10,79 @@ import {
     many2OneField,
 } from "@web/views/fields/many2one/many2one_field";
 import { UomAutoComplete } from "@uom/components/uom_autocomplete/uom_autocomplete";
+import { roundPrecision } from "@web/core/utils/numbers";
+import { onWillUpdateProps } from "@odoo/owl";
 
 export class Many2XUomTagsAutocomplete extends Many2XAutocomplete {
     static components = {
         ...Many2XAutocomplete.components,
         AutoComplete: UomAutoComplete,
     };
+    static props = {
+        ...Many2XAutocomplete.props,
+        productModel: { type: String, optional: true },
+        productId: { type: Number, optional: true },
+        productQuantity: { type: Number, optional: true },
+    };
+
+    async setup() {
+        super.setup();
+        onWillUpdateProps(async (nextProps) => {
+            if (nextProps.productModel !== this.props.productModel || 
+                nextProps.productId !== this.props.productId ||
+                nextProps.productQuantity !== this.props.productQuantity
+            ) {
+                await this.updateReferenceUnit(nextProps);
+                await this.render();
+            }
+        });
+        await this.updateReferenceUnit();
+    }
+
+    async render() {
+        await super.render();
+    }
+
+    async updateReferenceUnit(props = this.props) {
+        if (props.productModel && props.productId) {
+            const product = await this.orm.searchRead(props.productModel, [["id", "=", props.productId]], ["uom_id"]);
+            this.referenceUnit = (await this.orm.searchRead("uom.uom", [["id", "=", product[0].uom_id[0]]], ["name", "factor", "parent_path", "rounding"]))[0];
+        }
+    }
 
     async search(name) {
-        const records = await this.orm.searchRead(
+        let records = await this.orm.searchRead(
             this.props.resModel,
-            [...this.props.getDomain(), ["name", "ilike", name]], 
-            ["id", "name", "relative_factor", "relative_uom_id"],
+            [...this.props.getDomain(), ["name", "ilike", name]],
+            ["id", "name", "relative_factor", "factor", "relative_uom_id", "parent_path"],
         );
-        return records.map((record) => {
+        const hasCommonReference = (uom1, uom2) => {
+            const uom1Path = uom1.parent_path.split("/");
+            const uom2Path = uom2.parent_path.split("/");
+            const commonPath = [];
+            for (let i = 0; i < Math.min(uom1Path.length, uom2Path.length); i++) {
+                if (uom1Path[i] === uom2Path[i]) {
+                    commonPath.push(uom1Path[i]);
+                } else {
+                    break;
+                }
+            }
+            return commonPath.length > 0;
```

Comment:
If we simply want to check there is a common ancestor why not returning ? 
```javascript
return uom1Path[0] === uom2Path[0]
```

Reply:
Indeed, you're right. It's the same.